gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -969,6 +970,7 @@
   public static final Charset DEFAULT_CHARSET_ENCODING = Charsets.UTF_8;
   public static final String TEST_HARNESS_LAUNCHER_IMPL = "gobblin.testharness.launcher.impl";
   public static final int PERMISSION_PARSING_RADIX = 8;
+  public static final String NON_TRANSIENT_ERROR_MESSAGES = "non.transient.error.messages";
```

Comment:
Is this a new setting, or this property comes from the external system?

If it's a new one, I suggest to prefix it with "gobblin." , and consider a more descriptive name. Thing separate by dots usually mean class/property names. So in this case there will be class "non", class "transient", "error" and it will have a property "message". Something like "gobblin.errorMessages.logNonTransientErrors" could be more meaningful

Reply:
Also, you can add a purpose of this setting as a javadoc comment. I see that you included it in the commit message, but having it in code can be beneficial too, if people look at the list of settings and try to find out what they mean.

Reply:
It is a new one and describes a comma separated list of non transient error messages that can come in a gobblin job.

Reply:
Are those like exception classes or parts of the text in exception message? Would be good to add an example in javadoc alongside with intended use case.

Reply:
It can be either. Sometimes, the exception class is very generic, e.g. IOException, in that case this config can include error message also.
It all depends how to value of this config is being used. in li-gobblin, I compare it with exception.getMessage() which contains both class and message.